src/main/java/core/mate/academy/service/MachineProducer.java

```diff
@@ -1,11 +1,13 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Machine;
 import java.util.List;
 
-public interface MachineProducer {
+public interface MachineProducer<T extends Machine> {
```

Comment:
Parameterize `MachineProducer` correctly to ensure that only `Machine` or its subtypes can be used. This is done correctly here, as `T extends Machine` ensures that only `Machine` or its subtypes can be used.